framework/src/modules/chain/processor/processor.js

```diff
@@ -0,0 +1,329 @@
+		}
+		this.processors[processor.version] = processor;
+		this.matchers[processor.version] = matcher || (() => true);
+	}
+
+	// eslint-disable-next-line no-unused-vars,class-methods-use-this
+	async init(genesisBlock) {
+		this.logger.debug(
+			{ id: genesisBlock.id, payloadHash: genesisBlock.payloadHash },
+			'initializing processor',
+		);
+		// do init check for block state. We need to load the blockchain
+		const blockProcessor = this._getBlockProcessor(genesisBlock);
+		await this._processGenesis(genesisBlock, blockProcessor, {
+			skipSave: false,
+		});
+		await blockProcessor.init.run();
+		this.logger.info('Blockchain ready');
+	}
+
+	// process is for standard processing of block, especially when received from network
+	async process(block) {
+		return this.sequence.add(async () => {
+			this.logger.debug({ id: block.id }, 'starting to process block');
+			const blockProcessor = this._getBlockProcessor(block);
+			const { lastBlock } = this.blocksModule;
+
+			const forkStatus = await blockProcessor.forkStatus.run({
+				block,
+				lastBlock,
+			});
+
+			if (!forkStatusList.includes(forkStatus)) {
+				throw new Error('Unknown fork status');
+			}
+
+			// Discarding block
+			if (forkStatus === FORK_STATUS_IDENTICAL_BLOCK) {
+				this.logger.debug({ id: block.id }, 'Block already processed');
+				return;
+			}
+			if (forkStatus === FORK_STATUS_DISCARD) {
+				this.logger.debug({ id: block.id }, 'Discarding block');
+				return;
+			}
+			if (forkStatus === FORK_STATUS_DOUBLE_FORGING) {
+				this.logger.warn(
+					{ id: block.id, generatorPublicKey: block.generatorPublicKey },
+					'Discarding block due to double forging',
+				);
+				return;
+			}
+			// Discard block and move to different chain
+			if (forkStatus === FORK_STATUS_DIFFERENT_CHAIN) {
+				this.logger.debug({ id: block.id }, 'Detected different chain to sync');
+				this.channel.publish('chain:processor:sync');
+				return;
+			}
+			// Replacing a block
+			if (forkStatus === FORK_STATUS_TIE_BREAK) {
+				this.logger.info({ id: lastBlock.id }, 'Received tie breaking block');
+				await blockProcessor.validateNew.run({
+					block,
+					lastBlock,
+				});
+				await blockProcessor.validate.run({
+					block,
+					lastBlock,
+				});
+				const previousLastBlock = cloneDeep(lastBlock);
+				await this._deleteBlock(lastBlock, blockProcessor);
+				const newLastBlock = this.blocksModule.lastBlock;
+				try {
+					await this._processValidated(block, newLastBlock, blockProcessor);
+				} catch (error) {
+					this.logger.error(
+						{ id: block.id, previousBlockId: previousLastBlock.id, error },
+						'Failed to apply newly received block. restoring previous block.',
+					);
+					await this._processValidated(
+						previousLastBlock,
+						newLastBlock,
+						blockProcessor,
+						{ skipBroadcast: true },
+					);
+				}
+				return;
+			}
+
+			this.logger.debug(
+				{ id: lastBlock.id, height: block.height },
```

Comment:
Shouldn't this be `lastBlock.height`?

```suggestion
				{ id: lastBlock.id, height: lastBlock.height },
```

Reply:
I think in this case `id` is wrong. It should be `block.id`